modules/gapi/src/executor/gapi_tbb_executor.cpp

```diff
@@ -0,0 +1,362 @@
+                                {
+                                    ITT_AUTO_TRACE_GUARD(ittTbbEnqueueSpawnReadyBlocks);
+                                    auto start = timer.now();
+                                    arena.enqueue([ready_items, &root, body, unlock_root_wait_guard](){
+                                        parallel::batch_spawn(ready_items, root.get(), body);
+                                        //TODO: why we need this? Either write a descriptive comment or remove it
+                                        volatile auto p = unlock_root_wait_guard.guard.get();
+                                        util::suppress_unused_warning(p);
+                                    });
+                                    util::suppress_unused_warning(start);
+                                    LOG_DEBUG(NULL, "Enqued in "<< std::chrono::duration_cast<std::chrono::microseconds>(timer.now() - start).count() <<" mks \n");
+                                }
+                                //unlock master thread waiting on conditional variable (if any) to pick up enqueued tasks
+                                //wake master only if there is new work and there were no work before (i.e. root->ref_count() was 1 and master was
+                                //waiting on cv for async tasks to complete)
+                                block_master.unlock((new_root_ref_count == 2) ? wake_tbb_master::yes : wake_tbb_master::no);
+                            }
+                        };
+
+                        node->async_task_body(callback, node->total_order_index);
+                    }
+
+                    executed++;
+                    //reset dependecy_count to initial state
+                    node->dependency_count = node->dependencies;
+
+                    return result;
+                }
+
+                return parallel::use_tbb_scheduler_bypass::no;
+            };
+
+            auto num_start_tasks = q.size();
+
+            //TODO: use recursive spawning and task soft affinity for faster task distribution
+            //Spawn one less tasks and say TBB to reuse(recycle) current task.
+            //As graph is starting and there has no task been spawned yet
+            //assert_graph_is_running(root) will not hold, so spawn without assert
+            parallel::batch_spawn(num_start_tasks - 1,root.get(), body, /* assert_graph_is_running*/false);
+            root->spawn_and_wait_for_all(*parallel::allocate_task(root.get(), body));
+
+            std::chrono::high_resolution_clock timer;
+
+            auto tbb_work_done   = [&root]       (){ return root->ref_count() == 1; };
+            auto async_work_done = [&async_tasks](){ return 0 == async_tasks.count; };
+            do {
+//               //First participate in execution of TBB graph till there are no more ready tasks.
+               root->wait_for_all();
+
+               if (!async_work_done()) { //Bypass waiting on cv if there are no async work
```

Comment:
``there are no`` -> ``there is no``

Reply:
fixed

Reply:
>Seems that it is wrong place for //Bypass waiting on cv if there are no async work comment
Could you please elaborate why do you think it is wrong place for the comment ? (The comment itself explains why the check `!async_work_done()` is used - i.e. it is an optimization)

Reply:
`// Bypass waiting on cv if there no async work to do` -> `there is no`

Because this comment is in the opening brace of the logic block for the exact opposite operation

Reply:
changed